subsys/net/l2_wifi_if_conn/Kconfig

```diff
@@ -6,6 +6,7 @@
 menuconfig L2_WIFI_CONNECTIVITY
 	bool "L2 Wi-Fi Connectivity"
 	depends on NET_CONNECTION_MANAGER
+	default y if WPA_SUPP
```

Comment:
This is the one bit I am worried about. It would be nice if we could be more surgical about this.

> This means we can relax the current approach to automatically enable certain features a bit and move configuration to the overlay files.

I wonder if it would be acceptable to place `CONFIG_L2_WIFI_CONNECTIVITY=y` in the nrf700x overlay files

We have no intention of ever up-streaming those overlays, and furthermore the overlays won't function properly unless `CONFIG_L2_WIFI_CONNECTIVITY` winds up enabled, so it seems ***somewhat*** reasonable to enable `CONFIG_L2_WIFI_CONNECTIVITY` there.

Yes, this would mean that the nrf700x overlays would be referencing a config option from sdk-nrf, but since we don't intend to upstream those overlays, and since those overlays won't work unless used from within sdk-nrf, this seems like not such a big deal

Thoughts @rlubos @mbolivar-nordic ?

Reply:
> I don't really think, that someone would enable both connection manager and supplicant with no intention of using it for Wi-Fi management.

You're right, I missed that this specifically targets the supplicant and not the wifi driver as a whole. So my concerns are definitely moot.

I'd still probably consider it marginally cleaner to enable directly in the overlays, but you're right, there is truly not any issue with doing it the way you've done it so feel free to ignore that personal preference if you think it is best